--- ./main.c.orig	Tue Sep 30 22:53:04 2003
+++ ./main.c	Tue Sep 30 23:16:45 2003
@@ -82,6 +82,22 @@
 			dropbear_exit("Failed to set process group");
 		}
 		dropbear_log(LOG_INFO, "Running in background");
+#ifndef DEBUG_TRACE
+		{
+			int fd;
+			/* redirect stdin/stdout/stderr to /dev/null */
+			fd = open("/dev/null", O_RDWR);
+			if (fd != -1) {
+				dup2(fd, STDIN_FILENO);
+				dup2(fd, STDOUT_FILENO);
+				dup2(fd, STDERR_FILENO);
+				if (fd > 2) {
+					close(fd);
+				}
+			}
+		}
+#endif
+
 	}
 
 	/* create a PID file so that we can be killed easily */
